chrome/updater/updater_module.cc

```diff
@@ -98,17 +98,23 @@ void Observer::OnEvent(Events event, const std::string& id) {
       status = "Status is unknown.";
     } else if (crx_update_item_.state == ComponentState::kUpToDate &&
                updater_configurator_->GetPreviousUpdaterStatus().compare(
-                   updater_status_string_map.find(UpdaterStatus::kUpdated)
-                       ->second) == 0) {
+                   updater_status_string_map.at(UpdaterStatus::kUpdated)) == 0) {
       status = std::string(
-          updater_status_string_map.find(UpdaterStatus::kUpdated)->second);
+          updater_status_string_map.at(UpdaterStatus::kUpdated));
     } else {
       status = std::string(
           updater_status_string_map.find(status_iterator->second)->second);
     }
     if (crx_update_item_.state == ComponentState::kUpdateError) {
-      status +=
-          ", error code is " + std::to_string(crx_update_item_.error_code);
+      // QUICK_ROLL_FORWARD update, adjust the message to "Update installed,
+      // pending restart"
+      if (crx_update_item_.error_code == 21) {
+        status = std::string(
+            updater_status_string_map.at(UpdaterStatus::kUpdated));
+      } else {
+        status +=
+            ", error code is " + std::to_string(crx_update_item_.error_code);
```

Comment:
This is just going to add a number. Can we stringify these instead? https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/main:components/update_client/update_client_errors.h;l=74;drc=ac02e200288bef9c817fad9b06763463e4e9493d

Reply:
So the reason why we didn't stringify the error codes is because they're from multiple sources, not only from the update_client_errors.h file. They can be network errors or other errors. It's not always straightforward to stringify them.

Reply:
In that case, wouldn't just adding the number be ambiguous? IOW we should also log the `ErrorCategory` [1] to disambiguate the error enums entry, right? But honestly seeing an error like e.g. "Error category 2, error code 27" is really of little help (having had to face some of those myself), and creating this string is already creating more binary size. I'd still suggest fixing this situation (which is not your fault, the code is already in this situation) by adding a stringifier with those 2 parameters `ErrorCategory` and an `int` to identify the enum number (we can ignore that `enum class` entries technically cannot be casted to or from integers).


[1] https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/main:components/update_client/update_client_errors.h;l=27;drc=ac02e200288bef9c817fad9b06763463e4e9493d